### ISSUE TITLE
🐛 [firebase_analytics] I Can't remove the user id or the user properties for android.

### ISSUE THREAD

**Issue by julianjab** [FirebaseExtended/flutterfire#7442]
## Bug report

When I try to remove the user id/property using the plugin, it is not possible for android devices.
### Expected behavior
```dart
import 'package:firebase_analytics/firebase_analytics.dart';

class Analytics {
  FirebaseAnalytics analytics = FirebaseAnalytics();

  void setUserId(String? userId) {
    analytics.setUserId(userId);
  }

  void setUserProperty(String name, String? value) {
    analytics.setUserProperty(name: name, value: value);
  }
}


Analytics analytics = Analytics();

analytics.setUserId('12345'); // Add user ID 
analytics.setUserProperty('user_name', 'Jhon Doe') //Add user property "user_name"

//The error occurs when I try to remove that user ID or an user property for android Plugin i cpou
analytics.setUserId(null); // Should remove the user ID, but not works on android plugin. 
analytics.setUserProperty('user_name', null); // Should remove the user property "user_name", but not works on android plugin. 
```

## Additional context
Versions: 
[firebase_analytics: 8.1.2](https://pub.dev/packages/firebase_analytics/changelog)

Flutter plugin documentation.

[Setting a null [id] removes the user id(https://github.com/FirebaseExtended/flutterfire/blob/master/packages/firebase_analytics/firebase_analytics/lib/src/firebase_analytics.dart#L133)

[Setting a null [value] removes the user property.](https://github.com/FirebaseExtended/flutterfire/blob/master/packages/firebase_analytics/firebase_analytics/lib/src/firebase_analytics.dart#L189)

---

### Flutter doctor

Run `flutter doctor` and paste the output below:

<details><summary>Click To Expand</summary>

```
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 2.5.1, on macOS 12.0.1 21A559 darwin-x64, locale en)
[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
[✓] Xcode - develop for iOS and macOS
[✓] Chrome - develop for the web
[✓] Android Studio (version 2020.3)
[✓] VS Code (version 1.62.3)
[✓] Connected device (3 available)

• No issues found!
```

</details>

---

**Comment by darshankawar** [comment]
@julianjab 
Thanks for the report. I see that you are on a bit older version of stable and the plugin itself. Can you try upgrading both (stable to 2.5.3) and plugin to `^8.3.4` and see if it helps ?

**Comment by stx** [comment]
Same problem on Flutter stable 2.5.3 and firebase_analytics 9.0.2.

```
FirebaseAnalytics.instance.setUserId();
```

This throws an exception:

```
PlatformException(firebase_analytics, null, null, null)

StandardMethodCodec.decodeEnvelope (/Users/james/Development/flutter/packages/flutter/lib/src/services/message_codecs.dart:607)
MethodChannel._invokeMethod (/Users/james/Development/flutter/packages/flutter/lib/src/services/platform_channel.dart:156)
<asynchronous gap> (Unknown Source:0)
FirebaseAnalytics.setUserId (/Users/james/Development/flutter/.pub-cache/hosted/pub.dartlang.org/firebase_analytics-9.0.2/lib/src/firebase_analytics.dart:142)
<asynchronous gap> (Unknown Source:0)
```

**Comment by Salakar** [comment]
This was fixed in the 9.0.3 release - please try it out
